{"uid":"6d2945eccd452791","name":"Exception test","fullName":"tests.forgot_password_test.TestForgotPassword#test_expected_exception_on_page_title","historyId":"288cd0030c92ba34b4ff4b58ed8ec476","time":{"start":1713748626739,"stop":1713748636975,"duration":10236},"description":"Exception catching","descriptionHtml":"<p>Exception catching</p>\n","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5614fbb41863 <unknown>\n#1 0x5614fb8378c6 <unknown>\n#2 0x5614fb882618 <unknown>\n#3 0x5614fb8826d1 <unknown>\n#4 0x5614fb8c5744 <unknown>\n#5 0x5614fb8a45cd <unknown>\n#6 0x5614fb8c2c19 <unknown>\n#7 0x5614fb8a4343 <unknown>\n#8 0x5614fb875593 <unknown>\n#9 0x5614fb875f5e <unknown>\n#10 0x5614fbb0584b <unknown>\n#11 0x5614fbb097a5 <unknown>\n#12 0x5614fbaf3571 <unknown>\n#13 0x5614fbb0a332 <unknown>\n#14 0x5614fbad887f <unknown>\n#15 0x5614fbb30728 <unknown>\n#16 0x5614fbb308fb <unknown>\n#17 0x5614fbb409b4 <unknown>\n#18 0x7f85e0294ac3 <unknown>","statusTrace":"self = <tests.forgot_password_test.TestForgotPassword object at 0x7f4b83e21a00>\n\n    @allure.description(\"Exception catching\")\n    @allure.title(\"Exception test\")\n    @allure.link(\"github.com/allure-examples/\", name=\"Allure Examples\")\n    @allure.issue(\"github.com/allure-examples/allure-examples/issues/1\", name=\"ISSUE-1\")\n    @allure.testcase(\n        \"github.com/allure-examples/allure-examples/issues/2\", name=\"TESTCASE-2\"\n    )\n    def test_expected_exception_on_page_title(self):\n>       self.about_page.click_login_link()\n\ntests/forgot_password_test.py:53: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/about_page.py:20: in click_login_link\n    self.click(self.LOGIN_LINK)\npages/base_page.py:60: in click\n    el: WebElement = self.wait.until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"f12ead68dcd47833713574e6ad92d513\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7f4b83b6fba0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5614fbb41863 <unknown>\nE       #1 0x5614fb8378c6 <unknown>\nE       #2 0x5614fb882618 <unknown>\nE       #3 0x5614fb8826d1 <unknown>\nE       #4 0x5614fb8c5744 <unknown>\nE       #5 0x5614fb8a45cd <unknown>\nE       #6 0x5614fb8c2c19 <unknown>\nE       #7 0x5614fb8a4343 <unknown>\nE       #8 0x5614fb875593 <unknown>\nE       #9 0x5614fb875f5e <unknown>\nE       #10 0x5614fbb0584b <unknown>\nE       #11 0x5614fbb097a5 <unknown>\nE       #12 0x5614fbaf3571 <unknown>\nE       #13 0x5614fbb0a332 <unknown>\nE       #14 0x5614fbad887f <unknown>\nE       #15 0x5614fbb30728 <unknown>\nE       #16 0x5614fbb308fb <unknown>\nE       #17 0x5614fbb409b4 <unknown>\nE       #18 0x7f85e0294ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"session_request","time":{"start":1713748428127,"stop":1713748428127,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"base_url","time":{"start":1713748428127,"stop":1713748428127,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"_verify_url","time":{"start":1713748428127,"stop":1713748428127,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Exception catching","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5614fbb41863 <unknown>\n#1 0x5614fb8378c6 <unknown>\n#2 0x5614fb882618 <unknown>\n#3 0x5614fb8826d1 <unknown>\n#4 0x5614fb8c5744 <unknown>\n#5 0x5614fb8a45cd <unknown>\n#6 0x5614fb8c2c19 <unknown>\n#7 0x5614fb8a4343 <unknown>\n#8 0x5614fb875593 <unknown>\n#9 0x5614fb875f5e <unknown>\n#10 0x5614fbb0584b <unknown>\n#11 0x5614fbb097a5 <unknown>\n#12 0x5614fbaf3571 <unknown>\n#13 0x5614fbb0a332 <unknown>\n#14 0x5614fbad887f <unknown>\n#15 0x5614fbb30728 <unknown>\n#16 0x5614fbb308fb <unknown>\n#17 0x5614fbb409b4 <unknown>\n#18 0x7f85e0294ac3 <unknown>","statusTrace":"self = <tests.forgot_password_test.TestForgotPassword object at 0x7f4b83e21a00>\n\n    @allure.description(\"Exception catching\")\n    @allure.title(\"Exception test\")\n    @allure.link(\"github.com/allure-examples/\", name=\"Allure Examples\")\n    @allure.issue(\"github.com/allure-examples/allure-examples/issues/1\", name=\"ISSUE-1\")\n    @allure.testcase(\n        \"github.com/allure-examples/allure-examples/issues/2\", name=\"TESTCASE-2\"\n    )\n    def test_expected_exception_on_page_title(self):\n>       self.about_page.click_login_link()\n\ntests/forgot_password_test.py:53: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/about_page.py:20: in click_login_link\n    self.click(self.LOGIN_LINK)\npages/base_page.py:60: in click\n    el: WebElement = self.wait.until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"f12ead68dcd47833713574e6ad92d513\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7f4b83b6fba0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5614fbb41863 <unknown>\nE       #1 0x5614fb8378c6 <unknown>\nE       #2 0x5614fb882618 <unknown>\nE       #3 0x5614fb8826d1 <unknown>\nE       #4 0x5614fb8c5744 <unknown>\nE       #5 0x5614fb8a45cd <unknown>\nE       #6 0x5614fb8c2c19 <unknown>\nE       #7 0x5614fb8a4343 <unknown>\nE       #8 0x5614fb875593 <unknown>\nE       #9 0x5614fb875f5e <unknown>\nE       #10 0x5614fbb0584b <unknown>\nE       #11 0x5614fbb097a5 <unknown>\nE       #12 0x5614fbaf3571 <unknown>\nE       #13 0x5614fbb0a332 <unknown>\nE       #14 0x5614fbad887f <unknown>\nE       #15 0x5614fbb30728 <unknown>\nE       #16 0x5614fbb308fb <unknown>\nE       #17 0x5614fbb409b4 <unknown>\nE       #18 0x7f85e0294ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","steps":[{"name":"Click Login link","time":{"start":1713748626740,"stop":1713748636974,"duration":10234},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5614fbb41863 <unknown>\n#1 0x5614fb8378c6 <unknown>\n#2 0x5614fb882618 <unknown>\n#3 0x5614fb8826d1 <unknown>\n#4 0x5614fb8c5744 <unknown>\n#5 0x5614fb8a45cd <unknown>\n#6 0x5614fb8c2c19 <unknown>\n#7 0x5614fb8a4343 <unknown>\n#8 0x5614fb875593 <unknown>\n#9 0x5614fb875f5e <unknown>\n#10 0x5614fbb0584b <unknown>\n#11 0x5614fbb097a5 <unknown>\n#12 0x5614fbaf3571 <unknown>\n#13 0x5614fbb0a332 <unknown>\n#14 0x5614fbad887f <unknown>\n#15 0x5614fbb30728 <unknown>\n#16 0x5614fbb308fb <unknown>\n#17 0x5614fbb409b4 <unknown>\n#18 0x7f85e0294ac3 <unknown>\n\n","statusTrace":"  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/.venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/pages/about_page.py\", line 20, in click_login_link\n    self.click(self.LOGIN_LINK)\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/pages/base_page.py\", line 60, in click\n    el: WebElement = self.wait.until(\n                     ^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"public ip address","time":{"start":1713748636984,"stop":1713748637067,"duration":83},"status":"passed","steps":[],"attachments":[{"uid":"791af8371f446a28","name":"HTTP logs of http://checkip.amazonaws.com/","source":"791af8371f446a28.txt","type":"text/plain","size":359}],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"b8e0f1a4a13ee143","name":"Cookies","source":"b8e0f1a4a13ee143.json","type":"application/json","size":3637},{"uid":"27b17e684a954145","name":"Session Storage","source":"27b17e684a954145.json","type":"application/json","size":358},{"uid":"419878da263e5e0c","name":"Local Storage","source":"419878da263e5e0c.json","type":"application/json","size":350},{"uid":"15506a25049199a9","name":"Console Logs","source":"15506a25049199a9.json","type":"application/json","size":7723},{"uid":"9d11916ac18dd1bd","name":"Full Page Screenshot","source":"9d11916ac18dd1bd.png","type":"image/png","size":6238629},{"uid":"738df770e95c4df8","name":"URL","source":"738df770e95c4df8.uri","type":"text/uri-list","size":23},{"uid":"ee8800200c47739","name":"Network Logs","source":"ee8800200c47739.json","type":"application/json","size":916554}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":2,"attachmentsCount":8,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"session_request::0","time":{"start":1713748690846,"stop":1713748690846,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"parentSuite","value":"Custom parent suite"},{"name":"story","value":"Forgot Password Feature's Functionality"},{"name":"severity","value":"critical"},{"name":"epic","value":"Security"},{"name":"suite","value":"Custom suite"},{"name":"subSuite","value":"Custom sub suite"},{"name":"tag","value":"security"},{"name":"host","value":"fv-az1200-164"},{"name":"thread","value":"2395-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.forgot_password_test"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[{"name":"TESTCASE-2","url":"https://github.com/allure-examples/allure-examples/issues/2","type":"tms"},{"name":"ISSUE-1","url":"https://github.com/allure-examples/allure-examples/issues/1","type":"issue"},{"name":"Allure Examples","url":"https://github.com/allure-examples/","type":"link"}],"hidden":false,"retry":false,"extra":{"severity":"critical","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":["security"]},"source":"6d2945eccd452791.json","parameterValues":[]}